File: app/src/main/java/com/futnote/mobile/components/Note.java
package com.futnote.mobile.components;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "tblNote")
public class Note {

    @PrimaryKey(autoGenerate = true)
    private int id;
    private String judul;
    private String deskripsi;
    private int prioritas;

    public Note(int id, String judul, String deskripsi, int prioritas) {
        this.id = id;
        this.judul = judul;
        this.deskripsi = deskripsi;
        this.prioritas = prioritas;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getJudul() {
        return judul;
    }

    public void setJudul(String judul) {
        this.judul = judul;
    }

    public String getDeskripsi() {
        return deskripsi;
    }

    public void setDeskripsi(String deskripsi) {
        this.deskripsi = deskripsi;
    }

    public int getPrioritas() {
        return prioritas;
    }

    public void setPrioritas(int prioritas) {
        this.prioritas = prioritas;
    }
}

File: app/src/main/java/com/futnote/mobile/model/Note.kt
package com.futnote.mobile.model

import android.os.Parcel
import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "tblNote")
class Note(
        var judul: String,
        var deskripsi: String,
        var prioritas: Int): Parcelable {
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    constructor(parcel: Parcel) : this(
            parcel.readString().toString(),
            parcel.readString().toString(),
            parcel.readInt()) {
        id = parcel.readInt()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(judul)
        parcel.writeString(deskripsi)
        parcel.writeInt(prioritas)
        parcel.writeInt(id)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Note> {
        override fun createFromParcel(parcel: Parcel): Note {
            return Note(parcel)
        }

        override fun newArray(size: Int): Array<Note?> {
            return arrayOfNulls(size)
        }
    }
}